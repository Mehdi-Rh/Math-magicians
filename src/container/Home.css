
.Home {
  color: white;
  padding: 70px 0;
  margin: auto;
  text-align: center;
  width: 80vw;
}

.Home > * {
  margin: 1rem;
}
