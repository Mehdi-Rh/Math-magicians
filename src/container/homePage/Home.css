
.Home {
  color: white;
  padding: 70px 0;
  margin: calc(1vw + 0.5rem) 10vw;
  text-align: center;
}

.Home > h1 {
  font-size: xx-large;
  display: flex;
  margin: 1rem;
}

.text {
  margin: 1rem;
  font-size: x-large;
  display: flex;
  text-align: start;
}

@media screen and (max-width: 768px) {
  .Home {
    margin: 0.5rem 5vw;
    padding: 1rem 0;
  }

  .text {
    font-size: larger;
  }
}
