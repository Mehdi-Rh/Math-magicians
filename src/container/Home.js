import React from 'react';
import './Home.css';

const Home = () => {
  const text = ['In 1961 was born the first electronic calculator and for many years this invention was only available as a hard device format, by the time we started having it as a numeric application and this format becomes the most popular.', 'On this website, in the calculator section is integrated a numeric calculator where you can do your basic calculations ( add, subtract, multiply, devise, and modulo ).', 'As a bonus, a quotes section is added for further inspiration.'];
  return (
    <div className="Home">
      <h1>Welcome to our page!</h1>
      <div>
        <p>
          {text.map((p, index) => (
            <p key={p[index]} className="text">
              {p}
              <br />
            </p>
          ))}
        </p>
      </div>
    </div>
  );
};

export default Home;
