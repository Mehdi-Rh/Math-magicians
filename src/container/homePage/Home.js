import React from 'react';
import './Home.css';
import { fadeInRight } from 'react-animations';
import Radium, { StyleRoot } from 'radium';

const styles = {
  fadeInRightH1: {
    animation: 'x 1s',
    animationName: Radium.keyframes(fadeInRight, 'fadeInRight'),
  },
  fadeInRightP: {
    animation: 'x 2s',
    animationName: Radium.keyframes(fadeInRight, 'fadeInRight'),
  },
};

const Home = () => {
  const text = ['In 1961 was born the first electronic calculator and for many years this invention was only available in a hard device format, by the time we started having it as a numeric application and this format becomes the most popular.', 'On this website, in the calculator section is integrated a numeric calculator where you can do your basic calculations ( add, subtract, multiply, devise, and modulo ).', 'As a bonus, a quotes section is added for further inspiration.'];
  return (
    <StyleRoot>
      <div className="Home">
        <h1 style={styles.fadeInRightH1}>Welcome!</h1>
        <div style={styles.fadeInRightP}>
          <p>
            {text.map((p, index) => (
              <p key={p[index]} className="text">
                {p}
                <br />
              </p>
            ))}
          </p>
        </div>
      </div>
    </StyleRoot>
  );
};

export default Home;
