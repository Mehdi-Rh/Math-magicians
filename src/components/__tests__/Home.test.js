import React from 'react';
import Renderer from 'react-test-renderer';
import Home from '../../container/Quote';

it('renders correctly', () => {
  const tree = Renderer
    .create(<Home />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
