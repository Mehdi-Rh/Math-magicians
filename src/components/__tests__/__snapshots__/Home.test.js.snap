// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="quote"
>
  <p>
    Enjoy Five before Five
  </p>
  <p>
    Your Youth before your Old Age
  </p>
  <p>
    Your Health before your Illness
  </p>
  <p>
    Your Wealth before your Poverty
  </p>
  <p>
    Your Free Time before your Busyness
  </p>
  <p>
    and Your Life before your Death.
  </p>
  <p
    className="author"
  >
    Prophet Mohammed
  </p>
  <p
    className="author"
  >
    Peace Upon Him
  </p>
</div>
`;
